File: IndexGame/56mergeIntervals.py
#indexGame

# Definition for an interval.
# class Interval(object):
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution:
    def merge(self, intervals):
        intervals.sort(key = lambda x : x[0])
        res = []
        for i in intervals:
            if res and res[-1][1] >= i[0]:
                res[-1][1] = max(res[-1][1],i[1])
            else:
                res += [i]
        return res

File: IndexGame/56MergeIntervals.cpp
class Solution {
public:
    static bool comp(vector<int>& a, vector<int>& b){
        return a[0] < b[0];
    }
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> rev;
        if(intervals.size()==0) return rev;
        sort(intervals.begin(), intervals.end(), comp);
        rev.push_back(intervals[0]);
        for(int i=1; i<intervals.size(); i++){
            if(intervals[i][0]<=rev.back()[1]){
                vector<int> temp;
                temp.push_back(rev.back()[0]);
                temp.push_back(max(rev.back()[1], intervals[i][1]));
                rev.pop_back();
                rev.push_back(temp);
            }else{
                rev.push_back(intervals[i]);
            }
        }
        return rev;
    }
};